src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -1437,77 +1437,87 @@ namespace winrt::TerminalApp::implementation
         {
             if (const auto& realArgs = args.ActionArgs().try_as<SuggestionsArgs>())
             {
-                const auto source = realArgs.Source();
-                std::vector<Command> commandsCollection;
-                Control::CommandHistoryContext context{ nullptr };
-                winrt::hstring currentCommandline = L"";
-
-                // If the user wanted to use the current commandline to filter results,
-                //    OR they wanted command history (or some other source that
-                //       requires context from the control)
-                // then get that here.
-                const bool shouldGetContext = realArgs.UseCommandline() ||
-                                              WI_IsAnyFlagSet(source, SuggestionsSource::CommandHistory | SuggestionsSource::QuickFixes);
-                if (shouldGetContext)
-                {
-                    if (const auto& control{ _GetActiveControl() })
-                    {
-                        context = control.CommandHistory();
-                        if (context)
-                        {
-                            currentCommandline = context.CurrentCommandline();
-                        }
-                    }
-                }
+                _doHandleSuggestions(realArgs);
+                args.Handled(true);
+            }
+        }
+    }
 
-                // Aggregate all the commands from the different sources that
-                // the user selected. This is the order presented to the user
+    winrt::fire_and_forget TerminalPage::_doHandleSuggestions(SuggestionsArgs realArgs)
+    {
+        const auto source = realArgs.Source();
+        std::vector<Command> commandsCollection;
+        Control::CommandHistoryContext context{ nullptr };
+        winrt::hstring currentCommandline = L"";
+        winrt::hstring currentWorkingDirectory = L"";

```

Comment:
nit: always prefer `{}` for hstrings. no need to have empty string literals all over the place.

Reply:
I've previously raised a `nit` about how I personally think that `{}` is perfect for object constructions, while `=` is perfect for assignments, in particular of trivial types. I know you all like to write `auto foo{ bar() }`, so I feel quite betrayed seeing this!
(This is not a serious comment. 😄)